lib/propolis/src/block/file.rs

```diff
@@ -5,30 +5,64 @@
 use std::fs::{metadata, File, OpenOptions};
 use std::io::{Error, ErrorKind, Result};
 use std::num::NonZeroUsize;
+use std::os::raw::c_int;
+use std::os::unix::fs::FileTypeExt;
 use std::os::unix::io::AsRawFd;
 use std::path::Path;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
+use std::thread::JoinHandle;
 
 use crate::accessors::MemAccessor;
 use crate::block::{self, DeviceInfo};
+use crate::util::ioctl;
 use crate::vmm::{MappingExt, MemCtx};
 
 // XXX: completely arb for now
 const MAX_WORKERS: usize = 32;
 
+const DKIOC: i32 = 0x04 << 8;
+const DKIOCGETWCE: i32 = DKIOC | 36;
+const DKIOCSETWCE: i32 = DKIOC | 37;
+
 pub struct FileBackend {
     state: Arc<WorkerState>,
 
     worker_count: NonZeroUsize,
+    workers: Mutex<Vec<JoinHandle<()>>>,
 }
 struct WorkerState {
     attachment: block::backend::Attachment,
     fp: File,
 
+    wce_initial: Option<bool>,
+    wce_current: Mutex<Option<bool>>,
```

Comment:
also, kind of a nitpick, but throughout this file, we only ever refer to this setting with the acronym "WCE", so it's not that obvious to a reader who isn't familiar with ZFS that this stands for "Write-Cache Enable"...it might be nice if there was a comment either on these fields or on the `get_wce`/`set_wce` functions that expands the acronym?

Reply:
> This way, we don't need to handle the various cases where one of these is `Some` and the other is `None`, which (AFAICT) never actually happens?

I've rearranged things to resemble that.

> it might be nice if there was a comment either on these fields or on the get_wce/set_wce functions that expands the acronym?

Let me know if the comments are adequate.